lib/src/main/scala/nl.knaw.dans.easy.stage/lib/JSON.scala

```diff
@@ -34,19 +35,74 @@ object JSON extends DebugEnhancedLogging {
   val IS_MEMBER_OF_OAI_SET = "http://dans.knaw.nl/ontologies/relations#isMemberOfOAISet"
   val IS_SUBORDINATE_TO = "http://dans.knaw.nl/ontologies/relations#isSubordinateTo"
 
-  def createDatasetCfg(additionalLicenseFilenameAndMimetype: Option[(String, String)], audiences: Seq[String])(implicit s: Settings): Try[String] = Try {
+  def createDatasetCfg(additionalLicenseFilenameAndMimetype: Option[(String, String)],
+                       audiences: Seq[String],
+                       agreementsXmlExists: Boolean,
+                       messageFromDepositorExists: Boolean,
+                      )(implicit s: Settings): Try[String] = Try {
     trace(additionalLicenseFilenameAndMimetype, audiences)
 
     checkProvided("DOI", s.doi)
     checkProvided("URN", s.urn)
 
+    implicit def boolean2Opt(b: Boolean): Option[Unit] = {
+      Option(b).withFilter(identity).map(_ => ())
+    }
+
+    val bagMetadata = s.includeBagMetadata
+      .fold(List.empty[JObject])(_ => {
+        val mandatory = List(
+          ("contentFile" -> "dataset.xml") ~
+            ("dsID" -> "dataset.xml") ~
+            ("label" -> "DDM metadata for this dataset") ~
+            ("controlGroup" -> "X") ~
+            ("mimeType" -> "text/xml"),
+
+          ("contentFile" -> "files.xml") ~
+            ("dsID" -> "files.xml") ~
+            ("label" -> "File metadata for this dataset") ~
+            ("controlGroup" -> "X") ~
+            ("mimeType" -> "text/xml"),
+        )
+
+        val agreementsXmlEntry = agreementsXmlExists
+          .map(_ => {
+            ("contentFile" -> "agreements.xml") ~
+              ("dsID" -> "agreements.xml") ~
+              ("label" -> "Agreement specific metadata for this dataset") ~
+              ("controlGroup" -> "X") ~
+              ("mimeType" -> "text/xml")
+          })
+
+        val messageFromDepositorEntry = messageFromDepositorExists
+          .map(_ => {
+            ("contentFile" -> "message-from-depositor.txt") ~
+              ("dsID" -> "message-from-depositor.txt") ~
+              ("label" -> "Message from depositor to archivist about this dataset") ~
+              ("controlGroup" -> "M") ~
+              ("mimeType" -> "text/plain")
+          })
+
+        mandatory ++ agreementsXmlEntry ++ messageFromDepositorEntry
```

Comment:
Had a hard time understanding, perhaps rename
```suggestion
        mandatoryFiles ++ maybeAgreementsXmlEntry ++ maybeMessageFromDepositorEntry
```